packages/kbn-i18n/src/core/loader.js

```diff
@@ -0,0 +1,207 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ @typedef Messages - messages tree, where leafs are translated strings
+ @type {object<string, object>}
+ @property {string} [locale] - locale of the messages
+ @property {object} [formats] - set of options to the underlying formatter
+ */
+
+import path from 'path';
+import { readFile } from 'fs';
+import { promisify } from 'util';
+import { pick } from 'accept-language-parser';
+import JSON5 from 'json5';
```

Comment:
What do we need from json5 as opposed to regular old json?

Reply:
I'll let Maksim answer this question, but here is what we considered initially when json5 was chosen: https://github.com/elastic/kibana/tree/master/src/ui/ui_i18n#localization-files. For me the most important feature is the ability to have comments (e.g. to describe context or mark label as non-translatable) inside JSON.

Reply:
Yes, the main reason for using JSON5 is the ability to have comments inside JSON files.